web-common/src/features/sources/source-imported-utils.ts

```diff
@@ -13,12 +14,75 @@ export async function checkSourceImported(
     fileArtifacts.getFileArtifact(filePath).lastStateUpdatedOn;
   if (lastUpdatedOn) return; // For now only show for fresh sources
 
-  const success = await waitForResourceUpdate(
-    queryClient,
-    get(runtime).instanceId,
-    filePath,
-  );
+  waitForResourceUpdate(queryClient, get(runtime).instanceId, filePath)
+    .then((success) => {
+      console.log("s", success);
+      if (!success) return;
+      sourceImportedPath.set(filePath);
+    })
+    .catch(console.error);
+}
+
+function waitForResourceUpdate(
+  queryClient: QueryClient,
+  instanceId: string,
+  filePath: string,
+) {
+  return new Promise<boolean>((resolve) => {
+    const debouncer = createDebouncer();
+
+    const end = (changed: boolean) => {
+      unsub?.();
+      resolve(changed);
+    };
+
+    // eslint-disable-next-line prefer-const
+    const unsub = sourceImportedStore(
+      queryClient,
+      instanceId,
+      filePath,
+    ).subscribe(({ done, errored }) => {
+      if (!done) return;
 
-  if (!success) return;
-  sourceImportedPath.set(filePath);
+      debouncer(() => {
+        end(!errored);
+      }, 500);
```

Comment:
Why is the debounce necessary here?

Reply:
The source went into an IDLE state sometimes. With table check this is not be needed anymore.